src/http_connection.cpp

```diff
@@ -385,6 +385,7 @@ void http_connection::start(std::string const& hostname, int port
 				, std::bind(&http_connection::on_resolve
 				, me, _1, _2));
 		}
+		m_hostname = hostname;
```

Comment:
I don't follow this. Why is this change necessary?

Reply:
This is how this block of code was implemented before commit 592544f30f587294351bafce3dee8f53ff7cf03e.
This commit broke I2P support so most safe change in my opinion was to just revert it.

Probably, there is better way of fixing this problem, but I don't perfectly understand how this block of code works to make it better than it was before.

Reply:
so moving the assignment to `m_hostname` breaks i2p support?
by manual inspection, they look like they ought to have identical behavior

Reply:
Commit 592544f30f587294351bafce3dee8f53ff7cf03e puts assignment between `else` and `if`, which executes `if` regardless of what happens inside `#if TORRENT_USE_I2P` block.

Reply:
ouch. I see that now